### PR TITLE
Corrected AMSGrad max equation in Adam and AdamW

### LINKED ISSUE #142041: A typo in Adam docs
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.optim.Adam.html describes AMSGrad algorithm as follows:

amsgrad

The first equation should be:
correction

https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html also has the same issue.

### Suggest a potential alternative/fix

https://github.com/pytorch/pytorch/blob/eff99a4b4b056d165945f73ac9e0ba862928b597/torch/optim/adam.py#L296

It should be revised as the following: 
```
&\hspace{10mm}\widehat{v_t}^{max} \leftarrow \mathrm{max}(\widehat{v_{t-1}}^{max},
```

cc @svekars @brycebortree @sekyondaMeta @AlannaBurke @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by Uvi-12**
Fixes #142041

### CODE REVIEW

**Comment by Uvi-12** [comment]
@mikaylagawarecki I made corrections to the equation in all of the docs, please review it.

**Review by janeyx99 — APPROVED** [review]
Approving the optimizer docs changes.

Leaving the batchnorm docs to @mikaylagawarecki

**Comment by Uvi-12** [comment]
@janeyx99 Please approve the workflow runs and merge the PR.

**Comment by Uvi-12** [comment]
> Approving the optimizer docs changes.
> 
> 
> 
> Leaving the batchnorm docs to @mikaylagawarecki 

@The batchnorm docs PR is already reviewed and merged.

**Comment by Uvi-12** [comment]
@janeyx99 Please approve the workflows, and merge it.

**Comment by janeyx99** [comment]
@Uvi-12 ah you're likely going to have to rebase then.

**Review by janeyx99 — APPROVED** [review]
